### PR TITLE
Upgrade wrapper generator script and regenerate bindings

### PR DESCRIPTION
**Pull request by Gnimuc**
I'm testing https://github.com/JuliaInterop/Clang.jl/pull/278. Before making a new release for Clang.jl, I'd like to make sure that the code generated by the new generator is not broken. Any feedback about the new generator would be highly appreciated.

### CODE REVIEW

**Comment by rschwarz** [comment]
Note that the generation process was not totally automatic in our case, see #77 .

We also pinned `CEnum` to an older version (for no good reason except that the we didn't want to change the generated code) and later *vendored* the module.

**Comment by Gnimuc** [comment]
With this PR, the generation process should be totally automatic. 

Missing definitions are all macro-related and one can manually add them to `prologue.jl`. 

Variadic functions can be supported by using Julia 1.5's `@ccall` macro. In the future, Clang.jl should handle this, but for now, they're handled in `epilogue.jl`.

**Comment by Gnimuc** [comment]
> We also pinned `CEnum` to an older version (for no good reason except that the we didn't want to change the generated code) and later _vendored_ the module.

As SCIP doesn't use duplicated enum members, so I just completely remove the use of `CEnum.jl`. Clang.jl's new generator now has an option for this.

**Comment by rschwarz** [comment]
OK, so `LibSCIP.jl` is now the only file generated automatically by the script?
And the manual adjustments are all covered in `gen/prologue.jl`, `gen/epilogue.jl` or `src/wrapper.jl`?

That's a great improvement and will help a lot to keep up with API changes in SCIP in the future!

**Comment by matbesancon** [comment]
this is fantastic yes

File: gen/generator.toml [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+printer_blacklist = [
+    "printError",
+    "printErrorHeader"
+]
+
+[codegen]
+opaque_as_mutable_struct = false
+
+[codegen.macro]
+macro_mode = "basic"
```
Comment (matbesancon) [inline gen/generator.toml:18]:
```suggestion
macro_mode = "basic"

```

File: gen/generator.toml [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+library_name = "libscip"
+output_file_path = "../src/LibSCIP.jl"
+module_name = "LibSCIP"
+prologue_file_path = "./prologue.jl"
+epilogue_file_path = "./epilogue.jl"
+export_symbol_prefixes = ["SCIP_", "SCIP", "BMS_"]
+use_julia_native_enum_type = true
+printer_blacklist = [
+    "printError",
+    "printErrorHeader"
```
Comment (matbesancon) [inline gen/generator.toml:11]:
```suggestion
    "printErrorHeader",
```

**Comment by Gnimuc** [comment]
@matbesancon @rschwarz This is ready for another round of review. As for the test failures, I guess I didn't something wrong when solving the conflict. Feel free to edit this PR if needed. I'm going to merge the Clang.jl#278 this weekend and you could run the generator on your own machine.

**Comment by Gnimuc** [comment]
I won't keep updating this PR but will keep monitoring the compatibility of the generated code [here](https://github.com/Gnimuc/GeneratorScripts/tree/master/S/SCIP).

**Comment by matbesancon** [comment]
thanks for the work on this! So the current PR is based on what is currently on Clang.jl master right?

**Comment by Gnimuc** [comment]
Yes. It's recommended to use Clang.jl#master from now on.

**Comment by matbesancon** [comment]
there seems to be some conversion error: `unsafe_convert(::Type{Ptr{Ptr{Int8}}}, ::Base.RefValue{Cstring})`

**Comment by Gnimuc** [comment]
This has been fixed in https://github.com/scipopt/SCIP.jl/pull/186/commits/eba7eff535a5dab3d36d2793f9820b77ba6b9816, but I did something wrong when solving the conflict, so it's not in the latest commit. I believe you could fix this by cherry-pick that commit or just simply fix it with a new one.

**Comment by matbesancon** [comment]
@Gnimuc do you know when a release will be tagged on Clang.jl? Depending on master is a bit dangerous as next commits can break things

**Comment by Gnimuc** [comment]
This package is being monitored in https://github.com/Gnimuc/GeneratorScripts, it's safe to just use the master branch.

**Comment by Gnimuc** [comment]
Actually, this PR bundles the Modifest.toml, so it will always use the same version of Clang.jl unless someone does a `pkg> up`.

**Comment by matbesancon** [comment]
@Gnimuc should we bump the gen/Project.toml to the latest 0.11 you released?

**Comment by Gnimuc** [comment]
> @Gnimuc should we bump the gen/Project.toml to the latest 0.11 you released?

0.11? you mean 0.14? No, 0.14 has not been officially released yet. The latest master has some new features like variadic function support, docstrings dumping, etc. If you'd like to update to the latest master, run `pkg> up` in the project mode.

**Comment by matbesancon** [comment]
ok thank you, we do not need this for now so merging

**Pull request merged**